be/src/column/map_column.cpp

```diff
@@ -422,26 +422,40 @@ bool MapColumn::equals(size_t left, const starrocks::Column& rhs, size_t right)
         return false;
     }
 
+    bool has_null = false;
     for (uint32_t i = lhs_offset; i < lhs_end; ++i) {
         bool found = false;
         for (uint32_t j = rhs_offset; j < rhs_end; ++j) {
-            if (!_keys->equals(i, *(rhs_map._keys.get()), j)) {
+            int res = _keys->equals(i, *(rhs_map._keys.get()), j, safe_eq);
+            if (res == EQUALS_FALSE) {
                 continue;
             }
 
+            // key is null && isn't safe eq, don't need check value
+            has_null |= (res == EQUALS_NULL);
+            if (res == EQUALS_NULL && !safe_eq) {
+                found = true;
+                break;
+            }
```

Comment:
{null:2} == {null:3} => null when !safe_eq?

Reply:
yeah, I think the result is `NULL`. btw, trino don't support `NULL` as key

Reply:
but I think it should return false

Reply:
emmm, if return false that meanings key is safe equals?